### PR TITLE
Feature/23411 complete check function persist to pupilevent table storage

### PR DESCRIPTION
**Pull request by jon-shipley**
New complete check function to handle messages on the `check-complete`  queue put there by the Pupil SPA.

* sends the payload to the Admin DB
* updates the check status
* stores a copy of the message in `pupilEvents`
* deletes from the `preparedCheck` if not already deleted

### CODE REVIEW

File: admin/services/check-start.service.js [hunk 1]
```diff
@@ -236,10 +236,12 @@ checkStartService.prepareCheckQueueMessages = async function (checkFormAllocatio
       schoolPin: o.school_pin,
       pupilPin: o.pupil_pin,
       pupil: {
+        id: o.pupil_id,
```
Comment (GuyHarwood) [inline admin/services/check-start.service.js]: nice. i will remove this from the other story it was added to.

File: functions/lib/lib.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
+'use strict'
+
```
Comment (GuyHarwood) [inline functions/lib/lib.js]: can we call this file something more meaningful like `azure-storage-helper.js`?

in `index.js` above it is not clear what the intention is from requiring `lib/lib`

File: functions/completed-checks/index.js [hunk 3]
```diff
@@ -1,11 +1,60 @@
...
+// SQL server table
+const checkResultTable = '[checkResult]'
+const checkFormAllocationTable = '[checkFormAllocation]'
+const schema = ['mtc_admin']
+const checkStatusTable = '[checkStatus]'
+const checkTable = '[checkFormAllocation]'
 
-module.exports = function (context, completedCheckMessage) {
+// Table Storage
+const azureTableService = getPromisifiedAzureTableService()
```
Comment (GeorgeChatzigiannis) [inline functions/completed-checks/index.js]: @jon-shipley I still think promisifying the whole library for 2 method calls along with any potential internal dependency in other methods is too much.
The time it takes for the library 'promisification' to complete might be small but I believe it's unnecessary for the minimal usage you have.

File: functions/lib/lib.js (hunk 2 above)
Comment (jon-shipley) [reply functions/lib/lib.js]: Seems fair enough.  👍

File: functions/lib/lib.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    logger.error(error.message)
+    throw error
+  }
+}
+
+/**
+ * Promisify and cache the azureTableService library as it still lacks Promise support
+ */
+let azureTableService
+module.exports.getPromisifiedAzureTableService = function getPromisifiedAzureTableService () {
```
Comment (GeorgeChatzigiannis) [inline functions/lib/lib.js]: Perhaps here we could restrict to the methods that we want the loop to go through with regards to promisifying?

File: functions/completed-checks/index.js (hunk 3 above)
Comment (GeorgeChatzigiannis) [reply functions/completed-checks/index.js]: I am resolving this based on the answer given by Guy on #821

**Pull request merged**